server/service.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	pb "github-search-service/api"
+)
+
+type githubSearchServer struct {
+	pb.UnimplementedGithubSearchServiceServer
+}
+
+type GitHubItem struct {
+	HTMLURL string `json:"html_url"`
+	Repo    struct {
+		FullName string `json:"full_name"`
+	} `json:"repository"`
+}
+
+type GitHubResponse struct {
+	Items []GitHubItem `json:"items"`
+}
+
+func (s *githubSearchServer) Search(ctx context.Context, req *pb.SearchRequest) (*pb.SearchResponse, error) {
+
+	// Construct GitHub search query with optional user filter
+	query := req.SearchTerm
+	if req.User != "" {
+		query += " user:" + req.User
+	}
+
+	// Build GitHub API URL using the constructed query
+	url := fmt.Sprintf("https://api.github.com/search/code?q=%s", strings.ReplaceAll(query, " ", "+"))
+
+	// Prepare authenticated HTTP request to GitHub API
+	reqHTTP, _ := http.NewRequest("GET", url, nil)
```

Comment:
Better to build query params the native way:

```suggestion
	// Prepare authenticated HTTP request to GitHub API
	reqHTTP, _ := http.NewRequest("GET", "https://api.github.com/search/code", nil)

	q := url.Values{}
	q.Add("q", query)
	reqHTTP.URL.RawQuery = q.Encode()
```

Reply:
Also, need to handle the error here.